server/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -78,6 +79,11 @@ public interface PluginServices {
          */
         ClusterService clusterService();
 
+        /**
+         * A service to reroute shards to other nodes
+         */
+        RerouteService rerouteService();
+
```

Comment:
Out of curiosity, why PluginServices is an interface, not a record?

Reply:
An interface allows more flexibility for `Plugin` implementors for testing, mocking etc